server/sonar-search/src/main/java/org/sonar/search/EsSettings.java

```diff
@@ -149,6 +149,31 @@ private InetAddress readHost() {
     }
   }
 
+  private String readInitialStateTimeout() {
+    String initialStateTimeout = props.value(ProcessProperties.SEARCH_INITIAL_STATE_TIMEOUT, "30s").toLowerCase().trim();
+
+    if (initialStateTimeout.endsWith("ms") ||
+      initialStateTimeout.endsWith("s") ||
+      initialStateTimeout.endsWith("m") ||
+      initialStateTimeout.endsWith("h") ||
+      initialStateTimeout.endsWith("d") ||
+      initialStateTimeout.endsWith("w")
+      ) {
+      try {
+        int timeUnitLength = 1;
+        if (initialStateTimeout.endsWith("ms")) {
+          timeUnitLength = 2;
+        }
+        Double.parseDouble(initialStateTimeout.substring(0, initialStateTimeout.length() - timeUnitLength));
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png 'Severity: MAJOR') The return value of "parseDouble" must be used. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://next.sonarqube.com/sonarqube/coding_rules#rule_key=squid%3AS2201)